app/cdap/components/GridTable/index.tsx

```diff
@@ -0,0 +1,284 @@
+          const requestBody = directiveRequestBodyCreator(directives);
+          const sampleSpec = objectQuery(res, 'sampleSpec') || {};
+          const visualization = objectQuery(res, 'insights', 'visualization') || {};
+
+          const insights = {
+            name: sampleSpec.connectionName,
+            workspaceName: res.workspaceName,
+            path: sampleSpec.path,
+            visualization,
+          };
+          requestBody.insights = insights;
+
+          const workspaceUri = objectQuery(res, 'sampleSpec', 'path');
+          const workspaceInfo = {
+            properties: insights,
+          };
+          const gridParams = {
+            directives,
+            workspaceId,
+            workspaceUri,
+            workspaceInfo,
+            insights,
+          };
+          return MyDataPrepApi.execute(params, requestBody);
+        })
+      )
+      .subscribe((response) => {
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setWorkspace,
+          payload: {
+            data: response.values,
+            headers: response.headers,
+            types: response.types,
+            ...gridParams,
+          },
+        });
+        setLoading(false);
+        setGridData(response);
+      });
+  };
+
+  useEffect(() => {
+    // Get DATA from URL paramteres to get data of workspace
+    const payload = {
+      context: params.namespace,
+      workspaceId: params.wid,
+    };
+    getWorkSpaceData(payload, wid);
+  }, [wid]);
+
+  // ------------@createHeadersData Function is used for creating data of Table Header
+  const createHeadersData = (columnNamesList: string[], columnTypesList: IRecords) => {
+    if (Array.isArray(columnNamesList)) {
+      return columnNamesList.map((eachColumnName: string) => {
+        return {
+          name: eachColumnName,
+          label: eachColumnName,
+          type: [columnTypesList[eachColumnName]],
+        };
+      });
+    }
+  };
+
+  // ------------@convertNonNullPercent Function is used for calculation of Missing/Null value
+  const convertNonNullPercent = (nonNullValue) => {
+    const lengthOfData: number = gridData?.values.length;
+    let count: number = 0;
+    let emptyCount: number = 0;
+    let nullValueCount: number = 0;
+    if (lengthOfData) {
+      nullValueCount = nonNullValue.null ? (nonNullValue.null / 100) * lengthOfData : 0;
+      emptyCount = nonNullValue.empty ? (nonNullValue.empty / 100) * lengthOfData : 0;
+      count = parseInt(nullValueCount.toFixed(0) + emptyCount.toFixed(0));
+    }
+    return count;
+  };
+
+  // ------------@checkFrequentlyOccuredValues Function is used for checking which value appears maximum time in a column if that column doesn't have missing/null value
+  const checkFrequentlyOccuredValues = (key) => {
+    const valueOfKey = gridData.values.map((el) => el[key]);
+    let mostfrequentItem: number = 1;
+    let mostFrequentItemCount: number = 0;
+    let mostfrequentItemValue: string = '';
```

Comment:
Please make the capitalization of these variables consistent. (Prefer `mostFrequentItem...`)

Reply:
Done